app/battle/battle.py

```diff
@@ -0,0 +1,123 @@
+def battle(knightsConfig: dict):
+    # BATTLE PREPARATIONS:
+
+    # lancelot
+    lancelot = knightsConfig["lancelot"]
+
+    # apply armour
+    lancelot["protection"] = 0
+    for a in lancelot["armour"]:
+        lancelot["protection"] += a["protection"]
+
+    # apply weapon
+    lancelot["power"] += lancelot["weapon"]["power"]

```

Comment:
There is a potential issue here: if `lancelot['power']` is not initialized before this line, this operation will raise a `KeyError`. Ensure that `lancelot['power']` is initialized to a default value before this line.